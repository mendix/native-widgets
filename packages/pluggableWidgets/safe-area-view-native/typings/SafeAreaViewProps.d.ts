/**
 * This file was generated from SafeAreaView.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix Widgets Framework Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";

export interface SafeAreaViewProps<Style> {
    name: string;
    style: Style[];
    content?: ReactNode;
}

export interface SafeAreaViewPreviewProps {
    /**
     * @deprecated Deprecated since version 9.18.0. Please use class property instead.
     */
    className: string;
    class: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    content: { widgetCount: number; renderer: ComponentType<{ children: ReactNode; caption?: string }> };
}
