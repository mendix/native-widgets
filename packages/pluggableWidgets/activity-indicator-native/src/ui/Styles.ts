import { Style } from "@mendix/piw-native-utils-internal";
import { ViewStyle } from "react-native";

export interface ActivityIndicatorStyle extends Style {
    container: ViewStyle;
    indicator: {
        size: "large" | "small";
        color: string;
    };
}

export const defaultActivityStyle: ActivityIndicatorStyle = {
    container: {},
    indicator: {
        size: "large",
        color: "gray"
    }
};
