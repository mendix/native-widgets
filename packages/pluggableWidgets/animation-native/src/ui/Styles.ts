import { Style } from "@mendix/piw-native-utils-internal";
import { ViewStyle } from "react-native";

export interface AnimationStyle extends Style {
    container: ViewStyle;
}

export const defaultAnimationStyle: AnimationStyle = {
    container: {
        // All ViewStyle properties are allowed
    }
};
