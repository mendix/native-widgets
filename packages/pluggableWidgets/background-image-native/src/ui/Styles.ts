import { ViewStyle, ImageStyle } from "react-native";
import { Style } from "@mendix/piw-native-utils-internal";

interface CustomImageStyle extends ImageStyle {
    svgColor?: string;
}

export interface BackgroundImageStyle extends Style {
    container: ViewStyle;
    image: CustomImageStyle;
}

export const defaultBackgroundImageStyle: BackgroundImageStyle = {
    container: {},
    image: {}
};
