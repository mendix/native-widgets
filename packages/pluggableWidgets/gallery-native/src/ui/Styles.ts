import { ColorValue, TextStyle, ViewStyle } from "react-native";

export interface TouchableStyleProps {
    rippleColor?: ColorValue;
    borderless?: boolean;
    radius?: number;
    foreground?: boolean;
}

export interface TouchableStyle extends ViewStyle, TouchableStyleProps {}

export interface GalleryStyle {
    container?: ViewStyle;
    emptyPlaceholder?: ViewStyle;
    firstItem?: ViewStyle;
    lastItem?: ViewStyle;
    list?: ViewStyle;
    listItem?: ViewStyle;
    loadMoreButtonContainer?: ViewStyle;
    loadMoreButtonPressableContainer?: TouchableStyle;
    loadMoreButtonCaption?: TextStyle;
}

export const defaultGalleryStyle: GalleryStyle = {
    listItem: { flexGrow: 1 },
    loadMoreButtonContainer: {
        alignSelf: "stretch"
    },
    loadMoreButtonPressableContainer: {
        borderWidth: 1,
        borderStyle: "solid",
        rippleColor: "rgba(0, 0, 0, 0.2)",
        borderColor: "#264AE5",
        backgroundColor: "#264AE5",
        alignItems: "center",
        justifyContent: "center",
        borderRadius: 8,

        minWidth: 48,
        minHeight: 48,
        paddingVertical: 8,
        paddingHorizontal: 8
    },
    loadMoreButtonCaption: {
        color: "#FFFFFF",
        fontSize: 12,
        fontWeight: "bold",
        lineHeight: 14
    }
};
