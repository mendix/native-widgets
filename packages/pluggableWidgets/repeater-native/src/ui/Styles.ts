import { Style } from "@mendix/piw-native-utils-internal";
import { ViewStyle } from "react-native";

export interface RepeaterStyle extends Style {
    container: ViewStyle;
}

export const defaultRepeaterStyle: RepeaterStyle = {
    container: {}
};
