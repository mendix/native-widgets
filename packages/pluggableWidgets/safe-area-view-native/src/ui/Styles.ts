import { ViewStyle } from "react-native";
import { Style } from "@mendix/piw-native-utils-internal";

export interface SafeAreaViewStyle extends Style {
    container: ViewStyle;
}

export const defaultSafeAreaViewStyle: SafeAreaViewStyle = {
    container: {
        flex: 1
    }
};
