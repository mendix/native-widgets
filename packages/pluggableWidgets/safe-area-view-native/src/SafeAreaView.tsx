import { createElement } from "react";
import { flattenStyles } from "@mendix/piw-native-utils-internal";
import { SafeAreaViewStyle, defaultSafeAreaViewStyle } from "./ui/Styles";
import { SafeAreaViewProps } from "../typings/SafeAreaViewProps";
import { SafeAreaView as SafeAreaViewComponent } from "react-native-safe-area-context";
import { View } from "react-native";

export const SafeAreaView = (props: SafeAreaViewProps<SafeAreaViewStyle>): JSX.Element => {
    const styles = flattenStyles(defaultSafeAreaViewStyle, props.style);

    return (
        <View style={{ flex: 1, backgroundColor: styles.container.backgroundColor }}>
            <SafeAreaViewComponent style={{ flex: 1 }} pointerEvents={"box-none"} testID={props.name}>
                <View style={styles.container} pointerEvents={"box-none"}>
                    {props.content}
                </View>
            </SafeAreaViewComponent>
        </View>
    );
};
