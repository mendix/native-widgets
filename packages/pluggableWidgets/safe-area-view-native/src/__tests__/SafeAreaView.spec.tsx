import { createElement } from "react";
import { render } from "@testing-library/react-native";
import { SafeAreaView } from "../SafeAreaView";
import { SafeAreaViewProps } from "../../typings/SafeAreaViewProps";
import { SafeAreaViewStyle } from "../ui/Styles";
import { Text } from "react-native";
import { SafeAreaProvider } from "react-native-safe-area-context";

jest.mock("@react-navigation/bottom-tabs", () => ({
    useBottomTabBarHeight: jest.fn(() => 50)
}));

describe("Safe area view", () => {
    let defaultProps: SafeAreaViewProps<SafeAreaViewStyle>;

    beforeEach(() => {
        defaultProps = {
            name: "safe-area-view-test",
            style: [],
            content: <Text>Content</Text>
        };
    });

    it("renders with content", () => {
        const component = render(
            <SafeAreaProvider>
                <SafeAreaView name={defaultProps.name} style={defaultProps.style} content={defaultProps.content} />
            </SafeAreaProvider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it("renders without content", () => {
        delete defaultProps.content;
        const component = render(
            <SafeAreaProvider>
                <SafeAreaView {...defaultProps} />
            </SafeAreaProvider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it("renders with custom styling", () => {
        defaultProps.style = [
            {
                container: { backgroundColor: "blue" }
            },
            { container: { backgroundColor: "green" } }
        ];
        const component = render(
            <SafeAreaProvider>
                <SafeAreaView {...defaultProps} />
            </SafeAreaProvider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
