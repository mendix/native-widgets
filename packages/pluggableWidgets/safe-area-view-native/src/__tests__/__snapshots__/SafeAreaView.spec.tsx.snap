// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Safe area view renders with content 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "backgroundColor": undefined,
      "flex": 1,
    }
  }
  testID="safe-area-view-test"
>
  <View
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text>
      Content
    </Text>
  </View>
</View>
`;

exports[`Safe area view renders with custom styling 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "backgroundColor": "green",
      "flex": 1,
    }
  }
  testID="safe-area-view-test"
>
  <View
    pointerEvents="box-none"
    style={
      {
        "backgroundColor": "green",
        "flex": 1,
      }
    }
  >
    <Text>
      Content
    </Text>
  </View>
</View>
`;

exports[`Safe area view renders without content 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "backgroundColor": undefined,
      "flex": 1,
    }
  }
  testID="safe-area-view-test"
>
  <View
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  />
</View>
`;
